dask/array/tests/test_array_core.py

```diff
@@ -33,31 +33,54 @@
     from_array,
     from_delayed,
     from_func,
-    getem,
     getter,
+    graph_from_arraylike,
     normalize_chunks,
     optimize,
     stack,
     store,
 )
 from dask.array.utils import assert_eq, same_keys
 from dask.base import compute_as_if_collection, tokenize
-from dask.blockwise import broadcast_dimensions
+from dask.blockwise import Blockwise, broadcast_dimensions
 from dask.blockwise import make_blockwise_graph as top
 from dask.blockwise import optimize_blockwise
 from dask.delayed import Delayed, delayed
 from dask.utils import apply, ignoring, key_split, tmpdir, tmpfile
 from dask.utils_test import dec, inc
 
 
-def test_getem():
-    sol = {
-        ("X", 0, 0): (getter, "X", (slice(0, 2), slice(0, 3))),
-        ("X", 1, 0): (getter, "X", (slice(2, 4), slice(0, 3))),
-        ("X", 1, 1): (getter, "X", (slice(2, 4), slice(3, 6))),
-        ("X", 0, 1): (getter, "X", (slice(0, 2), slice(3, 6))),
-    }
-    assert getem("X", (2, 3), shape=(4, 6)) == sol
+def test_graph_from_arraylike():
+    d = 2
+    chunk = (2, 3)
+    shape = tuple(d * n for n in chunk)
+    arr = np.ones(shape)
+
+    def check_task(dsk, i, j):
+        key = ("X", i, j)
+        assert key in dsk
+
+        # Check the array is bound
+        func = dsk[key][0].dsk["X"][0]
```

Comment:
Asserting things about `SubgraphCallable`s is not simple. This drills pretty deeply into the object hierarchy and is certainly fragile. Would love to hear ideas about how to reason about this without this kind of logic (perhaps @rjzamora has done similar things?)